bokeh/core/property/instance.py

```diff
@@ -121,6 +127,25 @@ def _may_have_unstable_default(self):
         # because the instance value is mutable
         return self._default is not Undefined
 
+class InstanceDefault:
+    """ Provide a deferred initializer for Instance defaults.
+
+    This is useful for Bokeh models with Instance properties that should have
+    unique default values for every model instance. Using an InstanceDefault
+    will afford better user-facing documentation than a lambda initializer.
+
+    """
+    def __init__(self, model: Type[Model], **kwargs: Any) -> None:
```

Comment:
Right now it wouldn't help, because this PR doesn't deal with types, but in future this should be a generic and use param spec types from PEP 612.